app/src/main/java/pl/ludek/smat/home/pool_office_client/presentation/MainActivity.kt

```diff
@@ -25,105 +23,98 @@ class MainActivity : AppCompatActivity(), View.OnClickListener {
-    private lateinit var sensorDataTextView: TextView
-    private lateinit var relayFirstOnButton:Button
-    private lateinit var relayFiveOnButton: Button
-    private lateinit var relayAllOnButton: Button
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
-        relayFirstOnStr = getString(R.string.relay_1_on)
-        relayFirstOffStr = getString(R.string.relay_1_off)
-        relayFiveOnStr = getString(R.string.relay_5_on)
-        relayFiveOffStr = getString(R.string.relay_5_off)
-        relayAllOnStr = getString(R.string.relay_all_on)
-        relayAllOffStr = getString(R.string.relay_all_off)
-        errorRelayStr = getString(R.string.relay_error)
-        errorSensorStr = getString(R.string.data_from_sensor)
-        relayFirstOnButton = findViewById(R.id.relayOneOn)
-        relayFiveOnButton = findViewById(R.id.relayFieveOn)
-        relayAllOnButton = findViewById(R.id.relayAllOn)
-        dataFromRepository = ViewModelProvider(this).get(PoolInfoViewModel::class.java)
-        dataFromRelay = ViewModelProvider(this).get(RelayViewModel::class.java)
-        sensorDataTextView = findViewById(R.id.sensorDataView)
-
+        initView()
+        val binding = ActivityMainBinding.inflate(layoutInflater)
         val dataFromSensorObserver = Observer<PoolInfoData>{ inputDataFromSensor ->
             if(inputDataFromSensor.p1 == 0.0f && inputDataFromSensor.t1 == 0.0f && inputDataFromSensor.t2 == 0.0f && inputDataFromSensor.t3 == 0.0f){
-                    sensorDataTextView.text = errorSensorStr
+                 binding.sensorDataView.text = errorSensorStr
             }else {
                 val rez:String =" T1= " + inputDataFromSensor.t1 +
                         " T2= " + inputDataFromSensor.t2 +
                         " T3= " + inputDataFromSensor.t3 +
                         " P1= " + inputDataFromSensor.p1
-                sensorDataTextView.text = rez
+                binding.sensorDataView.text = rez
             }
         }
         dataFromSensor.currentSensorData.observe(this,dataFromSensorObserver)
         setDataToLiveData()
-        relayFirstOnButton.setOnClickListener(this)
-        relayFiveOnButton.setOnClickListener(this)
-        relayAllOnButton.setOnClickListener(this)
-    }
-
-     private fun setDataToLiveData(){
-         dataFromRepository.getDataFromLocalStorage().observe(this, Observer {
-             dataFromSensor.setPoolInfoData(it)
-         })
-     }
-
-    override fun onClick(v: View?) {
-        when(v?.id){
-            R.id.relayOneOn ->if(relayFirstOnButton.text == relayFirstOnStr){
-                    dataFromRelay.postOneRelayOn().observe(this, Observer {
-                    if(!it.errorRelay){ relayFirstOnButton.text = relayFirstOffStr
+        binding.relayOneOn.setOnClickListener {
+            if( binding.relayOneOn.text == relayFirstOnStr){
```

Comment:
This condition is very poor it require to store multiple redundant variables.
In the next step we will replace all the buttons with ToggleButtons and this problem will disappear,
but next time try to think of something else :)